source/windows/windows_pki_utils.c

```diff
@@ -123,12 +127,16 @@ int aws_load_cert_from_system_cert_store(const char *cert_path, HCERTSTORE *cert
         *cert_store, X509_ASN_ENCODING | PKCS_7_ASN_ENCODING, 0, CERT_FIND_HASH, &cert_hash, NULL);
 
     if (!*certs) {
+        WCHAR wszMsgBuff[512];
```

Comment:
This feels like a function, e.g. `bool aws_win_format_message(char *buffer, size_t buffer_size, DWORD lastError)`

Reply:
Yeah, consider a helper function that handles logging WindowsErrors. That makes it tougher to include a bunch of context in a single log statement, but it's ok if you log multiple times about one error. so like you have a helper function like:

```
// I always log: "Windows system function %s() failed with error code %d (%s)"
void s_log_win_err(const char *function_name, int win_err)
```

and you call that once to report the failed Windows call, then log again about the context it happened in. so the user's logs look like:
```
[ERROR] Windows system function CertOpenStore() failed with error code 5 (ACCESS_DENIED)
[ERROR] invalid certficate path /a/b/c/d. Failed to load cert store.
```